x-pack/test/security_solution_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common', 'timePicker', 'hosts', 'settings']);
+  const testSubjects = getService('testSubjects');
+  const esArchiver = getService('esArchiver');
+
+  describe('Endpoint Event Resolver', function () {
+    before(async () => {
+      await esArchiver.load('endpoint/resolver_tree', { useCreate: true });
+      await pageObjects.hosts.navigateToSecurityHostsPage();
+      const fromTime = 'Jan 1, 2018 @ 00:00:00.000';
+      const toTime = 'now';
+      await pageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await (await testSubjects.find('draggable-content-host.name')).click();
+      await testSubjects.existOrFail('header-page-title');
+      await (await testSubjects.find('navigation-events')).click();
+      await testSubjects.existOrFail('events-viewer-panel');
+      await testSubjects.exists('investigate-in-resolver-button', { timeout: 4000 });
+      await (await testSubjects.findAll('investigate-in-resolver-button'))[0].click();
```

Comment:
Just a quick note was looking over another PR by Dima and saw this comment about a similar line https://github.com/elastic/kibana/pull/77389#discussion_r488153757

Reply:
I do wait for it to appear before the click

Reply:
The comment was not about waiting for it to appear but about not referencing the element in the array.  Instead of 

`return (await testSubjects.findAll('policyActionsButton'))[0];`

He recommended using:
`return await testSubjects.find('policyActionsButton');`

Here's his comment:
> testSubjects.find wraps driver.findElement that will return the first element in DOM with specified locator and throw exception in case it is not in DOM. It is better than searching for an array and selecting by index when array can be empty.

Reply:
But then it will not click I think I try that already

Reply:
Ok, I'm not too worried about it but his pr was to increase reliability.